pyat/at/lattice/variable_elements.py

```diff
@@ -68,74 +89,103 @@ def __init__(self, family_name, AmplitudeA=None, AmplitudeB=None,
 
         .. note::
 
-            * For all excitation modes all least one amplitude has to be provided.
+            * If no parameters are given it will be initialized as IdentityPass.
+            * For all excitation modes at least one amplitudes (A or B) has
+              to be provided.
               The default excitation is ``ACMode.SINE``
             * For ``mode=ACMode.SINE`` the ``Frequency(A,B)`` corresponding to the
-              ``Amplitude(A,B)`` has to be provided
+              ``Amplitude(A,B)`` has to be provided.
             * For ``mode=ACMode.ARBITRARY`` the ``Func(A,B)`` corresponding to the
-              ``Amplitude(A,B)`` has to be provided
+              ``Amplitude(A,B)`` has to be provided.
         """
-        kwargs.setdefault('PassMethod', 'VariableThinMPolePass')
-        self.MaxOrder = kwargs.pop('MaxOrder', 0)
-        self.Periodic = kwargs.pop('Periodic', True)
-        self.Mode = int(mode)
-        if AmplitudeA is None and AmplitudeB is None:
-            raise AtError('Please provide at least one amplitude for A or B')
-        AmplitudeB = self._set_params(AmplitudeB, mode, 'B', **kwargs)
-        AmplitudeA = self._set_params(AmplitudeA, mode, 'A', **kwargs)
-        self._setmaxorder(AmplitudeA, AmplitudeB)
-        if mode == ACMode.WHITENOISE:
-            self.Seed = kwargs.pop('Seed', datetime.now().timestamp())
-        self.PolynomA = numpy.zeros(self.MaxOrder+1)
-        self.PolynomB = numpy.zeros(self.MaxOrder+1)
-        ramps = kwargs.pop('Ramps', None)
-        if ramps is not None:
-            assert len(ramps)==4, \
-                'Ramps has to be a vector with 4 elements'
-            self.Ramps = ramps
-        super(VariableMultipole, self).__init__(family_name, **kwargs)
-
-    def _setmaxorder(self, ampa, ampb):
+        if len(kwargs) > 0:
+            self.FamName = family_name
+            if "AmplitudeA" not in kwargs and "AmplitudeB" not in kwargs:
+                raise AtError("Please provide at least one amplitude for A or B")
+            # start setting up Amplitudes and modes
+            # fist modes are called differently
+            modepyatinput = kwargs.pop("mode", ACMode.SINE)
+            modefromdict = kwargs.pop("Mode", None)
+            if modefromdict is not None:
+                mode = modefromdict  # matlab class
+            else:
+                mode = modepyatinput
+            self.Mode = int(mode)
+            # MaxOrder is later overwritten by lenth of amplitudes
+            self.MaxOrder = kwargs.get("MaxOrder", 0)
+            amplitudea = None
+            amplitudeb = None
+            if "AmplitudeA" in kwargs:
```

Comment:
This lol ne and the ones below can be removed

Reply:
What do you mean by lol ?